internal/collectors/logs_test.go

```
@@ -1,12 +1,95 @@
+// build the string manually to match real syslog output.
+func syslogTS(t time.Time) string {
+	return fmt.Sprintf("%s %2d %02d:%02d:%02d",
+		t.Month().String()[:3], t.Day(), t.Hour(), t.Minute(), t.Second())
+}
+
+func TestGetLogsUnitFilter(t *testing.T) {
+	t.Parallel()
+	logDir := t.TempDir()
+	ts := syslogTS(time.Now())
+
+	if err := os.WriteFile(filepath.Join(logDir, "syslog"), []byte(ts+" host kernel[1]: boot message\n"), 0o644); err != nil {
```

golangci-lint (2.12.2)

[medium] G306: Expect WriteFile permissions to be 0600 or less

(gosec)